From 0d8a77d76b67f4f530483d10cdf4d67d9c1479e6 Mon Sep 17 00:00:00 2001
Message-Id: <0d8a77d76b67f4f530483d10cdf4d67d9c1479e6.1428642548.git.chang-joon.lee@intel.com>
In-Reply-To: <bf1030973d60050117965176cfa2e18cdaf9c0a1.1428642548.git.chang-joon.lee@intel.com>
References: <bf1030973d60050117965176cfa2e18cdaf9c0a1.1428642548.git.chang-joon.lee@intel.com>
From: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
Date: Thu, 26 Feb 2015 21:59:16 +0530
Subject: [PATCH 2/7] FOR_UPSTREAM [VPG]: drm/i915: Populate aspect ratio in
 AVI Infoframe

Populate aspect ratio parameter of AVI Info Frame from aspect ratio of mode.

Issue: GMINL-6518
Change-Id: I73c063013796affea40889186c380cc4377dda56
Signed-off-by: Akashdeep Sharma <akashdeep.sharma@intel.com>
Signed-off-by: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
---
 drivers/gpu/drm/drm_edid.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/drm_edid.c b/drivers/gpu/drm/drm_edid.c
index f766219..d7a013a 100644
--- a/drivers/gpu/drm/drm_edid.c
+++ b/drivers/gpu/drm/drm_edid.c
@@ -3791,10 +3791,13 @@ drm_hdmi_avi_infoframe_from_display_mode(struct hdmi_avi_infoframe *frame,
 
 	frame->picture_aspect = HDMI_PICTURE_ASPECT_NONE;
 
-	/* Populate picture aspect ratio from CEA mode list */
-	if (frame->video_code > 0)
+	if  (mode->picture_aspect_ratio == HDMI_PICTURE_ASPECT_4_3
+		|| mode->picture_aspect_ratio == HDMI_PICTURE_ASPECT_16_9) {
+		frame->picture_aspect = mode->picture_aspect_ratio;
+	} else if (frame->video_code > 0) {
 		frame->picture_aspect = drm_get_cea_aspect_ratio(
 						frame->video_code);
+	}
 
 	frame->active_aspect = HDMI_ACTIVE_ASPECT_PICTURE;
 	frame->scan_mode = HDMI_SCAN_MODE_UNDERSCAN;
-- 
1.7.9.5

